File: src/Reporting.Core/Models/ReportDetailsModel.cs
namespace Reporting.Core.Models
{
    public class ReportDetailsModel
    {
        public ReportModel? Report { get; set; }
        public IEnumerable<ReportParameterModel>? Parameters { get; set; }
        public IEnumerable<ReportColumnDefinitionModel> ColumnDefinitions { get; set; } = Array.Empty<ReportColumnDefinitionModel>();
    }
}

File: src/reporting-app/src/models/report-details-model.ts
import { ReportParameterModel } from './report-parameter-model';
import { ReportColumnDefinitionModel } from './report-column-definition-model';
import { ReportModel } from './report-model';

export interface ReportDetailsModel {
    report: ReportModel;
    parameters?: ReportParameterModel[];
    columnDefinitions: ReportColumnDefinitionModel[];
}
